cpp/src/parquet/column_writer.cc

```diff
@@ -1043,7 +1043,7 @@ Status ConvertDictionaryToDense(const ::arrow::Array& array, MemoryPool* pool,
 }
 
 static inline bool IsDictionaryEncoding(Encoding::type encoding) {
-  return encoding == Encoding::PLAIN_DICTIONARY;
+  return encoding == Encoding::PLAIN_DICTIONARY || encoding == Encoding::RLE_DICTIONARY;
```

Comment:
Can we remove this function and instead adding a function to return if the ColumnWriter uses dictionary encoding?